src/Middleware/RateLimiting/src/RateLimitingMetrics.cs

```diff
@@ -0,0 +1,153 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using System.Runtime.CompilerServices;
+using Microsoft.Extensions.Metrics;
+
+namespace Microsoft.AspNetCore.RateLimiting;
+
+internal sealed class RateLimitingMetrics : IDisposable
+{
+    public const string MeterName = "Microsoft.AspNetCore.RateLimiting";
+
+    private readonly Meter _meter;
+    private readonly UpDownCounter<long> _currentLeaseRequestsCounter;
+    private readonly Histogram<double> _leaseRequestDurationCounter;
+    private readonly UpDownCounter<long> _currentRequestsQueuedCounter;
+    private readonly Histogram<double> _queuedRequestDurationCounter;
+    private readonly Counter<long> _leaseFailedRequestsCounter;
+
+    public RateLimitingMetrics(IMeterFactory meterFactory)
+    {
+        _meter = meterFactory.CreateMeter(MeterName);
+
+        _currentLeaseRequestsCounter = _meter.CreateUpDownCounter<long>(
+            "current-lease-requests",
+            description: "Number of HTTP requests that are currently active on the server that hold a rate limiting lease.");
```

Comment:
I'm not familiar with the area but I assume these strings don't need localization?

Reply:
Display name for event counters don't. This is new stuff but I'll guess no. Can change easily enough if we need to.